src/hydrabus/hydrabus_mode_onewire.c

```diff
@@ -180,58 +215,149 @@ uint8_t onewire_read_u8(t_hydra_console *con)
 	return value;
 }
 
-void onewire_scan(t_hydra_console *con)
+static uint8_t onewire_crc8(uint8_t value, struct onewire_scan_state *state)
 {
-	uint8_t id_bit_number = 0;
-	uint8_t last_zero = 0;
-	uint8_t id_bit = 0, cmp_id_bit = 0;
-	uint8_t search_direction = 0;
-	uint8_t LastDiscrepancy = 0;
-	uint8_t LastDeviceFlag = 0;
-	uint8_t i;
-	uint8_t ROM_NO[8] = {0};
-	onewire_start(con);
-	onewire_write_u8(con, 0xf0);
-	cprintf(con, "Discovered devices : ");
-	while(!LastDeviceFlag && !hydrabus_ubtn()) {
-		do{
+	state->crc8 = onewire_crc_table[state->crc8 ^ value];
+
+	return state->crc8;
+}
+
+static bool onewire_search(t_hydra_console *con, struct onewire_scan_state *state, enum onewire_scan_mode mode)
+{
+	int id_bit_number;
+	int last_zero, rom_byte_number, search_result;
+	int id_bit, cmp_id_bit;
+	unsigned char rom_byte_mask, search_direction;
+
+	/* Initialize global search state.  */
+	if(mode == onewire_scan_start) {
+		state->last_discrepancy = 0;
+		state->last_device_p = false;
+		state->last_family_discrepancy = 0;
+	}
+
+	/* Initialize for this search.  */
+	id_bit_number = 1;
+	last_zero = 0;
+	rom_byte_number = 0;
+	rom_byte_mask = 1;
+	search_result = 0;
```

Comment:
Once defined as `bool`, change to 
`search_result = false;`

Reply:
Made that a bool.